### PR TITLE
Centralize DDP speedups in docs

### LINKED ISSUE #11698: Centralize DDP optimizations in docs
## 📚 Documentation

DDP optimizations for `find_unused_parameters` and tuning NCCL parameters are listed here: https://pytorch-lightning.readthedocs.io/en/latest/guides/speed.html#when-using-ddp-plugins-set-find-unused-parameters-false

However, there's another section on the doc which explicitly talks about DDP optimizations here: https://pytorch-lightning.readthedocs.io/en/latest/advanced/advanced_gpu.html#ddp-optimizations

The pieces from the speed guide should be moved to the DDP optimizations section, and then the speed guide can simply refer to the advanced GPU training section for pointers. This way users have one consistent way to get specifics on how to tune DDP settings for speedups.

And the speedup doc can be a higher-level index of what users can do to explore faster training across dataloading/accelerators/strategies and general guidance for writing efficient pytorch programs

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda @rohitgr7 @justusschock @kaushikb11 @awaelchli @akihironitta

### PR DESCRIPTION
**Pull request by ananthsub**
## What does this PR do?

Fixes #11698

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [n/a] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/guides/speed.rst [hunk 1]
```diff
@@ -77,53 +77,7 @@ Whereas :class:`~pytorch_lightning.strategies.ddp.DDPStrategy` only performs two
...
-
-* `NCCL_NSOCKS_PERTHREAD <https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/env.html#nccl-nsocks-perthread>`__
-* `NCCL_SOCKET_NTHREADS <https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/env.html#nccl-socket-nthreads>`__
-* `NCCL_MIN_NCHANNELS <https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/env.html#nccl-min-nchannels>`__
-
-.. code-block:: bash
-
-    export NCCL_NSOCKS_PERTHREAD=4
-    export NCCL_SOCKET_NTHREADS=2
+For more details on how to tune performance with DDP, please see the :ref:`DDP Optimizations <ddp-optimizations>` section.
```
Comment (krshrimali) [inline docs/source/guides/speed.rst:80]: Thanks for adding a note here :)

**Review by krshrimali — APPROVED** [review]
This LGTM. Thanks, @ananthsub! 🚀

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: docs/source/advanced/model_parallel.rst [hunk 2]
```diff
@@ -718,6 +718,73 @@ DDP Optimizations
 ^^^^^^^^^^^^^^^^^
 
 
+When Using DDP Strategies, Set find_unused_parameters=False
```
Comment (awaelchli) [inline docs/source/advanced/model_parallel.rst:721]: This is unfortunately using the same title level as the super header for "DDP Optimizations".
